### ISSUE TITLE
more of a question than an issue, i am not seeing all of the columns that are in the data set in the output of the biologic.mprfile data 

### ISSUE THREAD

**Issue by oliverfernihough** [echemdata/galvani#114]
when i process a file using the MPR -> MPT function in the BT-Lab software, i get these columns: 

mode	ox/red	error	control changes	Ns changes	counter inc.	Ns	I Range	time/s	control/V/mA	Ecell/V	I/mA	dq/mA.h	(Q-Qo)/mA.h	|Energy|/W.h	freq/Hz	|Z|/Ohm	Phase(Z)/deg	Q charge/discharge/mA.h	half cycle	Temperature/ C	z cycle	Energy charge/W.h	Energy discharge/W.h	Capacitance charge/ F	Capacitance discharge/ F	step time/s	Re(Z)/Ohm	-Im(Z)/Ohm	Re(Y)/Ohm-1	Im(Y)/Ohm-1	|Y|/Ohm-1	Phase(Y)/deg	x	Q discharge/mA.h	Q charge/mA.h	Capacity/mA.h	Efficiency/%	control/V	control/mA	cycle number	P/W	R/Ohm	

however when i use the galvani MPRfile.data in the following code 

mpr_data = BioLogic.MPRfile(filepath)
df = pd.DataFrame(mpr_data.data).columns

print(df)

the output is: 

'flags', 'Ns', 'I Range', 'time/s', 'control/V/mA', 'Ewe/V', 'I/mA',
       'dq/mA.h', '(Q-Qo)/mA.h', '|Energy|/W.h', 'freq/Hz', '|Z|/Ohm',
       'Phase(Z)/deg', 'Q charge/discharge/mA.h', 'half cycle',
       'Temperature/°C', 'z cycle'
       
i just wanted to know if i was doing something wrong ?  or is it just a slight difference between the BT-Lab, EC-Lab versions ? 

i can generate most of the columns i need from the data but i just wondered if i didn't need to ?

**Comment by ml-evs** [comment]
I don't think you're doing anything wrong; BT-Lab is a new thing with more functionality than galvani knows about; at some point we may be able to support new columns but for now they have to be added manually.

**Comment by oliverfernihough** [comment]
> I don't think you're doing anything wrong; BT-Lab is a new thing with more functionality than galvani knows about; at some point we may be able to support new columns but for now they have to be added manually.

im pretty new to reading binary, but i've seen the structure in the MPdata_colID_dtype_map, in theory if i just add to those columns i will find what the new columns are called ?

**Comment by oliverfernihough** [comment]
closed the other issue to move the responses here: 

i think the data types were apparent, but if they're wrong i guess i can just cycle through the options to try and determine them 

thanks for the advice i will post some of the column names i find and their Dtype so they can be added to a later update

**Comment by oliverfernihough** [comment]
i've been looking through the binary, 

im not seeing some of the columns that are available in the MPT version of the file, I am wondering does anyone think that the export to text feature in the software, is actually applying certain calculations to the data to generate those columns and hence they're not stored in the binary ? 

just wondering if anyone else has noticed this

**Comment by PeterKraus** [comment]
Yes, that's correct: e.g. [cycle numbers](https://github.com/dgbowl/yadg/issues/150) are often computed by EC-Lab and not stored in the `mpr` files, and perhaps also [other things](https://github.com/dgbowl/yadg/issues/41).
